mesonbuild/interpreter/interpreter.py

```diff
@@ -781,7 +781,7 @@ def run_command_impl(self,
                     progname = name
                     break
             else:
-                raise MesonBugException('cmd was a built executable but not found in overrides table')
+                raise InterpreterException(f'Program {cmd.name!r} is an executable and therefore cannot be used during configuration')
```

Comment:
BTW don't you think it would be helpful to explain why exactly the build executable can't be used during configuration.

Meaning that configure step happens before any executables are build so any system executable can be used but not build targets.

Reply:
Added that it is a compiled executable.